hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -533,13 +539,17 @@ namespace hpx { namespace threads { namespace detail
 
             // if nothing else has to be done either wait or terminate
             else {
-                ++idle_loop_count;
+                HPX_ASSERT(next_thrd == nullptr);
 
-                if (scheduler.SchedulingPolicy::wait_or_add_new(
-                        num_thread, running, idle_loop_count))
+                bool running = this_state.load() < state_stopping;
```

Comment:
Once merged, this would be duplicated (see https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/runtime/threads/detail/scheduling_loop.hpp#L343).